hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1530,6 +1530,11 @@ protected void stopServiceThreads() {
 
     LOG.debug("Stopping service threads");
 
+    // stop procedure executor priority to other services such as server manager and assignment
```

Comment:
Do you mean 'prior to other services'; i.e. before? If so, s/priority/prior/.

Reply:
Is this a synchronous call?

Reply:
It will wait until all the worker threads are quit.